### PR TITLE
Add literature background convnet

### PR DESCRIPTION
**Pull request by lilianabs**
Issue [#340](https://github.com/FluxML/model-zoo/issues/340)

### CODE REVIEW

File: vision/conv_mnist/conv_mnist.jl [hunk 1]
```diff
@@ -149,15 +210,31 @@ function train(; kws...)
...
+# The function `train` performs the following tasks:
+
+# * Checks whether there is a GPU available and uses it for training the model. Otherwise, it uses the CPU.
+# * Loads the MNIST data using the function `get_data`.
+# * Creates the model and uses the [ADAM optimiser](https://fluxml.ai/Flux.jl/stable/training/optimisers/#Flux.Optimise.ADAM) with weight decay.
+# * Loads the [TensorBoardLogger.jl](https://github.com/JuliaLogging/TensorBoardLogger.jl) for logging data to Tensorboard.
+# * Creates the function `report` for computing the loss and accuracy during the training loop. It outputs these values to the TensorBoardLogger.
+# * Runs the training loop using [Flux’s training routine](https://fluxml.ai/Flux.jl/stable/training/training/#Training). For each epoch (step), it executes the following:
+#   * Computes the model’s predictions.
+#   * Computes the loos.
```
Comment (CarloLucibello) [inline vision/conv_mnist/conv_mnist.jl:230]:
```suggestion
#   * Computes the loss.
```

File: vision/conv_mnist/conv_mnist.jl [hunk 2]
```diff
@@ -1,7 +1,39 @@
...
+
+# * A convolutional encoder consisting of two convolutional layers.
+# * A dense block consisting of three fully-connected layers.
+
+# The basic units in each convolutional block are a convolutional layer, a sigmoid activation function, 
+# and a subsequent average pooling operation. Each convolutional layer uses a 5×5 kernel and a sigmoid activation function. 
+# These layers map spatially arranged inputs to a number of two-dimensional feature maps, typically increasing the number of channels. 
+# The first convolutional layer has 6 output channels, while the second has 16. 
+# Each 2×2 pooling operation (stride 2) reduces dimensionality by a factor of 4 via spatial downsampling. 
+# The convolutional block emits an output with shape given by (batch size, number of channel, height, width).
```
Comment (CarloLucibello) [inline vision/conv_mnist/conv_mnist.jl:16]:
```suggestion
# The convolutional block emits an output with shape given by (width, height, number of channels,  batch size).
```

File: vision/conv_mnist/conv_mnist.jl (hunk 1 above)
Comment (CarloLucibello) [reply vision/conv_mnist/conv_mnist.jl:230]: Also maybe this block of comments should be on top of the `train` definition

**Comment by ToucheSir** [comment]
Is this good to go?

**Comment by lilianabs** [comment]
Yes, I think is ready to merge :)

**Review by ToucheSir — APPROVED** [review]

**Pull request merged**